presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1172,21 +1173,27 @@ private static Domain buildColumnDomain(ColumnHandle column, List<HivePartition>
     @Override
     public Optional<ConnectorNewTableLayout> getInsertLayout(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
-        HivePartitionResult hivePartitionResult = partitionManager.getPartitions(metastore, tableHandle, Constraint.alwaysTrue());
-        if (!hivePartitionResult.getBucketHandle().isPresent()) {
+        HiveTableHandle hiveTableHandle = (HiveTableHandle) tableHandle;
```

Comment:
We definitely need to do those checks. I think they should be done `HiveMetadata.getTableHandle()`, so that we never return a table handle for an inaccessible table and thus we protect all code paths (all metadata operations for a specific table should operate on a handle).

Let's move those checks to a utility in `MetastoreUtil`, then call it from `getTableHandle`. We can also call the code from `HiveWriteUtils` (the `checkWritable` method basically duplicates the checks).